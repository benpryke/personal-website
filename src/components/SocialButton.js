import React from 'react';

/**
 * A link to social media
 */
export default class SocialButton extends React.Component {
  render() {
    const { name, url, icon } = this.props;

    return (
      <a href={url} className='social-button'>
        <i className={icon}></i>{name}
      </a>
    );
  }
}

export const LinkedInButton = () => <SocialButton name='LinkedIn' icon='fab fa-linkedin' url='https://linkedin.com/in/benjaminpryke'/>;
export const TwitterButton = () => <SocialButton name='Twitter' icon='fab fa-twitter' url='https://twitter.com/BenjaminPryke'/>;
export const GitHubButton = () => <SocialButton name='GitHub' icon='fab fa-github' url='https://github.com/benpryke'/>;
export const StackOverflowButton = () => <SocialButton name='Stack Overflow' icon='fab fa-stack-overflow' url='https://stackoverflow.com/users/604687/ninjakannon'/>;
export const InstagramButton = () => <SocialButton name='Instagram' icon='fab fa-instagram' url='https://www.instagram.com/gym_ninja_ben'/>;
