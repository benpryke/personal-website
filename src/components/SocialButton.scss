.social-button {
  display: inline-block;
  margin: 10px;
  font-size: 1rem;
  font-weight: 500;
  padding: 3px 10px;
  border-radius: 5px;
  border: none;
  background: $primary-colour-light;
  text-decoration: none;
  color: $primary-text-colour;
  white-space: nowrap;
  
  &:focus {
    animation: wave 0.2s ease-in;
  }
  
  &:hover svg {
    color: $primary-colour;
    transition: all 0.2s ease;
  }
  
  i {
    margin-right: 6px;
  }
}

// Button click animation
@keyframes wave {
  0% {
    box-shadow: 0 0 0 0 rgba($primary-colour, 0);
  }

  50% {
    box-shadow: 0 0 0 3px rgba($primary-colour, 0.33);
  }

  100% {
    box-shadow: 0 0 0 4px rgba($primary-colour, 0);
  }
}
