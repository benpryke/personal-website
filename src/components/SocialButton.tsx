import React from 'react';

export interface SocialButtonProps {
  /** Name of the social network */
  name: string,
  /** URL to link to */
  url: string,
  /** FontAwesome icon class name */
  icon: string,
}

/**
 * A link to social media
 */
export default class SocialButton extends React.Component<SocialButtonProps> {
  render() {
    const { name, url, icon } = this.props;

    return (
      <a href={url} className='social-button'>
        <i className={icon}></i>{name}
      </a>
    );
  }
}

export const LinkedInButton = () => <SocialButton name='LinkedIn' icon='fab fa-linkedin' url='https://linkedin.com/in/benjaminpryke'/>;
export const TwitterButton = () => <SocialButton name='Twitter' icon='fab fa-twitter' url='https://twitter.com/BenjaminPryke'/>;
export const GitHubButton = () => <SocialButton name='GitHub' icon='fab fa-github' url='https://github.com/benpryke'/>;
export const StackOverflowButton = () => <SocialButton name='Stack Overflow' icon='fab fa-stack-overflow' url='https://stackoverflow.com/users/604687/ninjakannon'/>;
export const InstagramButton = () => <SocialButton name='Instagram' icon='fab fa-instagram' url='https://www.instagram.com/gym_ninja_ben'/>;

export const socialButtons = {
  LinkedInButton,
  TwitterButton,
  GitHubButton,
  StackOverflowButton,
  InstagramButton,
};
