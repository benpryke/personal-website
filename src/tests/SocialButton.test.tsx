import React from 'react';
import { faLinkedin } from '@fortawesome/free-brands-svg-icons';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import SocialButton, { SocialButtonProps, socialButtons } from '../components/SocialButton';

const components = {
  SocialButton,
  ...socialButtons,
};

const props: SocialButtonProps = {
  name: 'name',
  url: 'url',
  icon: faLinkedin,
};

Object.entries(components).forEach(([name, Component]) => {
  describe(name, () => {
    it('renders without crashing', () => {
      mount(<Component {...props}/>);
    });

    it('should match the snapshot', () => {
      const tree = renderer.create(<Component {...props}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
