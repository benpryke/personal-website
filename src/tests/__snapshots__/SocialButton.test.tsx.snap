// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubButton should match the snapshot 1`] = `
<a
  className="social-button"
  href="https://github.com/benpryke"
>
  <i
    className="fab fa-github"
  />
  GitHub
</a>
`;

exports[`InstagramButton should match the snapshot 1`] = `
<a
  className="social-button"
  href="https://www.instagram.com/gym_ninja_ben"
>
  <i
    className="fab fa-instagram"
  />
  Instagram
</a>
`;

exports[`LinkedInButton should match the snapshot 1`] = `
<a
  className="social-button"
  href="https://linkedin.com/in/benjaminpryke"
>
  <i
    className="fab fa-linkedin"
  />
  LinkedIn
</a>
`;

exports[`SocialButton should match the snapshot 1`] = `
<a
  className="social-button"
  href="url"
>
  <i
    className="icon"
  />
  name
</a>
`;

exports[`StackOverflowButton should match the snapshot 1`] = `
<a
  className="social-button"
  href="https://stackoverflow.com/users/604687/ninjakannon"
>
  <i
    className="fab fa-stack-overflow"
  />
  Stack Overflow
</a>
`;

exports[`TwitterButton should match the snapshot 1`] = `
<a
  className="social-button"
  href="https://twitter.com/BenjaminPryke"
>
  <i
    className="fab fa-twitter"
  />
  Twitter
</a>
`;
